### ISSUE TITLE
[Impeller] TextureSourceVK::SetLayout API is hard to use correctly.

### ISSUE THREAD

**Issue by chinmaygarde** [flutter/flutter#147751]
If the new layout is the same as the old layout, the barrier is never inserted. This may cause issues if the transition is elided. And the caller may depend on the barrier being inserted. The method must return if a transition was actually performed.

**Comment by jonahwilliams** [comment]
We should probably just always insert the barrier?

**Comment by chinmaygarde** [comment]
Yeah, that would work but I am not sure about the performance implications. Likely minor. If there was an fml::Status<T> where T was an enum, we could return a status that said the layout was already correct.
